src/main/java/io/jenkins/plugins/appcenter/task/internal/CheckFileExistsTask.java

```diff
@@ -35,13 +35,15 @@
         final PrintStream logger = taskListener.getLogger();
         final CompletableFuture<Boolean> future = new CompletableFuture<>();
 
-        final FilePath remotablePath = filePath.child(request.pathToApp);
         try {
-            if (remotablePath.exists()) {
-                logger.println(String.format("File found: %s", request.pathToApp));
+
+            FilePath[] remoteFiles = filePath.list(request.pathToApp);
+
+            if (remoteFiles != null && remoteFiles.length != 0) {
```

Comment:
I don't know if I mentioned it but I also have a branch that I am working on that adds this functionality and it looks very similar to yours. I think we'll take your branch forward and get this merged but I want to bring in some of the changes from my branch and add them here.

In my version of this file I add additional checks along the lines of:

* List the files matching the pattern
* If the number of files is greater than 1 then complete exceptionally
* If the number of files is less than 1 then complete exceptionally
* If the number of files is one complete

Reply:
I've updated the file to include the above changes.